tempo_embeddings/embeddings/vector_database.py

```diff
@@ -0,0 +1,336 @@
+        pass
+
+    @abstractmethod
+    def retrieve_vectors_if_exist(self, collection: Collection, passages: List[Passage]):
+        pass
+
+
+class ChromaDatabaseManager(VectorDatabaseManagerWrapper):
+    """A Chroma Database can create N collections and will always use THE SAME embedder function and tokenizer for all collections.
+        To create collections with different embedders one needs to create separate databases
+    """
+    def __init__(self, 
+                 db_path: str = "default_db",
+                 embedder_name: str = "sentence-transformers/all-MiniLM-L6-v2",
+                 embedder_config: Dict[str,Any] = None,
+                 batch_size: int = 8):
+        super().__init__(batch_size)
+        self.db_path = db_path
+        self.embedder_name = embedder_name
+        self.embedder_config = embedder_config
+        self.tokenizer = AutoTokenizer.from_pretrained(embedder_name)
+        self.client =  None
+        self.requires_explicit_embeddings = False
+        if self.embedder_config and self.embedder_config.get("type") == "hf":
+            self.embedding_function = embedding_functions.HuggingFaceEmbeddingFunction(
+                api_key=self.embedder_config.get("api_key"),
+                model_name=embedder_name
+            )
+        elif self.embedder_config and self.embedder_config.get("type") == "custom":
+            self.requires_explicit_embeddings = True
+            self.embedding_function = None
+        else:
+            self.embedding_function = None
+
+        self.config = {
+            "embedder_name": self.embedder_name,
+            "embedder_type": self.embedder_config["type"],
+            "requires_explicit_embeddings": self.requires_explicit_embeddings,
+            "existing_collections": []
+        }
+    
+    def _save_config(self):
+        if self.config:
+            json.dump(self.config, open(f"{self.db_path}/config.json", "w"), indent=2)
+
+    def connect(self):
+        if not self.client:
+            self.client = chromadb.PersistentClient(path=self.db_path)
+            # If the given path existed then we will load the records AND override the parameters with the config
+            config_path = f"{self.db_path}/config.json"
+            if os.path.exists(config_path):
+                self.config = json.load(open(config_path))
+                print(f"Database Path '{self.db_path}' already exists. Loading current configuration\n{self.config}")
+            else:
+                print(f"Creating NEW Database in {self.db_path}...")
+                self._save_config()
+        else:
+            print(f"A connection to the client already exsists in this session: {self.client}")
+
+    def create_new_collection(self, name: str, passages: List[Passage] = None, embeddings: ArrayLike = None) -> Optional[Collection]:
+        if not self.client:
+            print("Please connect to a valid database first")
+            return None
```

Comment:
I think I would raise an exception here because this should never happen intentionally.

```suggestion
            raise RuntimeError("Please connect to a valid database first")
```